roles/openstack_undercloud/tasks/main.yml

```diff
@@ -48,15 +48,31 @@
 
 
 # ansible_env.HOME should've worked here
-- name: Get Controller Nodes number
-  shell: source {{ user_homedir.stdout }}/stackrc; nova list | grep controller | grep ACTIVE | wc -l
-  register: controller_count
-  when: undercloud_conf.stat.exists
-
-- name: Get Compute Nodes number
-  shell: source {{ user_homedir.stdout }}/stackrc; nova list | grep compute | grep ACTIVE | wc -l
-  register: compute_count
-  when: undercloud_conf.stat.exists
+- name: get overcloud node count before OSP-17
+  block:
+    - name: Get Controller Nodes number
+      shell: source {{ user_homedir.stdout }}/stackrc; nova list | grep controller | grep ACTIVE | wc -l
+      register: controller_count
+      when: undercloud_conf.stat.exists
```

Comment:
would be better to move this condition too along with the rhosp_major block condition

Reply:
done